### PR TITLE
Fix bug in pipeline_runs COUNT() query

### PR DESCRIPTION
**Pull request by reductionista**
Also added 2 tests for it. This was untested before, since it only gets
    run if the job you're searching for was not one of the most recent 1000
    job runs to complete.

### CODE REVIEW

File: core/services/job/job_orm_test.go [hunk 1]
```diff
@@ -985,7 +985,7 @@ func Test_FindPipelineRunIDsByJobID(t *testing.T) {
 	}
 
 	// Creation of job runs above cannot run in parallel, otherwise run ids are unpredictable
-	t.Parallel()
+	//t.Parallel()
```
Comment (jmank88) [inline core/services/job/job_orm_test.go:988]: Remove or restore?

File: core/services/job/orm.go [hunk 2]
```diff
@@ -903,19 +903,19 @@ func (o *orm) loadPipelineRunIDs(jobID *int32, offset, limit int, tx pg.Queryer)
...
-					),
+						`SELECT COUNT(p.id) FROM pipeline_runs AS p %s p.id >= $1 AND p.id <= $2`, filter,
+					), minID, maxID,
 				)
 				if err != nil {
 					err = errors.Wrap(err, "error loading from pipeline_runs")
 					return
 				}
-				offset -= skipped
+				offset -= skipped[0]
```
Comment (samsondav) [inline core/services/job/orm.go:918]: Why this change to slice?

**Review by samsondav — CHANGES_REQUESTED** [review]

File: core/services/job/orm.go (hunk 2 above)
Comment (reductionista) [reply core/services/job/orm.go:918]: It has to be a slice, because it gets passed as the destination output for `Select()`, which returns a list of rows.  But glad you pointed this out, because it occurs to me now, we could just use `Get()` instead which would be cleaner.  I'll go ahead and change that!

File: core/services/pipeline/orm.go [hunk 3]
```diff
@@ -490,7 +493,7 @@ WHERE pipeline_runs.id = batched_pipeline_runs.id`,
 
 	deleteTS := time.Now()
 
-	o.lggr.Debugw("pipeline_runs reaper DELETE query completed", "duration", deleteTS.Sub(start))
+	o.lggr.Debugw("pipeline_runs reaper DELETE query completed", "rows_deleted", rowsDeleted, "duration", deleteTS.Sub(start))
```
Comment (samsondav) [inline core/services/pipeline/orm.go:496]:
```suggestion
	o.lggr.Debugw("pipeline_runs reaper DELETE query completed", "rowsDeleted", rowsDeleted, "duration", deleteTS.Sub(start))
```

**Review by samsondav — CHANGES_REQUESTED** [review]
Very nice

**Review by samsondav — APPROVED** [review]
